lib/guardian/plug/ensure_not_authenticated.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Guardian.Plug.EnsureNotAuthenticated do
+  @moduledoc """
+  This plug ensures that a invalid JWT was provided and has been
+  verified on the request.
+
+  If one is not found, the `authenticated/2` function is invoked with the
+  `Plug.Conn.t` object and its params.
+
+  ## Example
+
+      # Will call the authenticated/2 function on your handler
+      plug Guardian.Plug.EnsureAuthenticated, handler: SomeModule
+
+      # look in the :secret location.  You can also do simple claim checks:
+      plug Guardian.Plug.EnsureAuthenticated, handler: SomeModule, key: :secret
+
+      plug Guardian.Plug.EnsureAuthenticated, handler: SomeModule, aud: "token"
+
+  If the handler option is not passed, `Guardian.Plug.ErrorHandler` will provide
+  the default behavior.
+  """
```

Comment:
@hassox @doomspork Do we want to default the error handler to the same error handler that `Guardian.Plug.EnsureAuthenticated` uses? It seems like it might get confusing using the same error handler handling when the user is unauthenticated and authenticated. Also, if we go this route, `Guardian.Plug.ErrorHandler` will need a new `authenticated/2` method.